src/middlewared/middlewared/plugins/datastore/write.py

```diff
@@ -57,7 +57,10 @@ async def insert(self, name: str, data: dict, options: dict | None = None):
         insert, relationships = self._extract_relationships(table, options['prefix'], data)
         for column in table.c:
             if column.default is not None:
-                insert.setdefault(column.name, column.default.arg)
+                value = column.default.arg
+                if callable(value):
```

Comment:
Instead of allowing any type of callable, do you think we should define a class that developers can use to try and enforce some sort of uniformity?

Reply:
SQLAlchemy does this check for us, you can’t just pass any callable there, it checks the signature to ensure that it accepts 0 or 1 arguments

Reply:
Okay so why do `value(None)`?

Reply:
Huh.  FWIW, I **_happened_** to experiment with similar code yesterday, and when I tried `column.default.arg()` it complained about a missing `ctx`.

Reply:
@yocalebo `column.default.arg` is a SQLAlchemy wrapper that always accepts some context (default value callables may accept some context). We’re not in an ORM, so we don’t have any context; hence, `None`